tools/release_packages

```diff
@@ -0,0 +1,372 @@
+from contextlib import contextmanager
+import os
+import argparse
+import glob
+import shutil
+import tempfile
+
+from git import Repo
+import crayons
+import twine.commands.upload
+import setuptools.sandbox
+
+REPO_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+repo = Repo(REPO_DIR)
+git = repo.git
+
+@contextmanager
+def cd(newdir):
+    prevdir = os.getcwd()
+    os.chdir(os.path.expanduser(newdir))
+    try:
+        yield
+    finally:
+        os.chdir(prevdir)
+
+def _generate_dist(dist_type, setup_file, package_name, setup_args):
+    message = 'Generating {dist_type} for {package_name}.'.format(
+        dist_type=dist_type,
+        package_name=package_name,
+    )
+    print(crayons.white(message, bold=True))
+
+    setup_dir = os.path.dirname(setup_file)
+    with cd(setup_dir):
+        setuptools.sandbox.run_setup(setup_file, setup_args)
+
+    message = '{dist_type} for {package_name} generated under {dir}.\n'.format(
+        dist_type=dist_type,
+        package_name=package_name,
+        dir=setup_dir,
+    )
+    print(crayons.green(message, bold=True))
+
+def generate_sdist(setup_file, package_name):
+    _generate_dist('sdist', setup_file, package_name, ['sdist'])
+
+def generate_bdist_wheel_universal(setup_file, package_name):
+    _generate_dist('bdist_wheel', setup_file, package_name,
+                   ['bdist_wheel', '--universal'])
+
+def twine_upload(dist_dirs):
+    message = 'Uploading distributions under the following directories:'
+    print(crayons.green(message, bold=True))
+    for dist_dir in dist_dirs:
+        print(crayons.yellow(dist_dir))
+    twine.commands.upload.main(dist_dirs)
+
+def cleanup(package_dir):
+    build_dir = os.path.join(package_dir, 'build')
+    temp_dir = os.path.join(package_dir, 'temp')
+    dist_dir = os.path.join(package_dir, 'dist')
+    egg_info = os.path.join(
+        package_dir,
+        '{}.egg-info'.format(os.path.basename(package_dir))
+    )
+    version_symlink = os.path.join(package_dir, 'version.py')
+
+    def _rm_if_it_exists(directory):
+        if os.path.isdir(directory):
+            print(crayons.green('Removing {}/*.'.format(directory), bold=True))
+            shutil.rmtree(directory)
+
+    map(_rm_if_it_exists, [build_dir, temp_dir, dist_dir, egg_info])
+
+    if os.path.islink(version_symlink):
+        print(crayons.green('Removing {}.'.format(version_symlink), bold=True))
+        os.remove(version_symlink)
+
+def increment_version(init_file, new_version):
+    fh, temp_abs_path = tempfile.mkstemp()
+    with os.fdopen(fh, 'w') as new_file, open(init_file) as old_file:
+        for line in old_file:
+            if line.startswith('__version__'):
+                prev = line.split('=')[-1].strip().replace('"', '')
+                new_file.write(line.replace(prev, new_version))
+            else:
+                new_file.write(line)
+
+    os.remove(init_file)
+    shutil.move(temp_abs_path, init_file)
```

Comment:
Can we reuse this substitution code snippet for `set_git_repo_var`?

Reply:
Hmm.. we could, yes! I'll get that done! Thanks, Robert! :)